mongo/datadog_checks/mongo/mongo.py

```diff
@@ -224,17 +263,20 @@ def _authenticate(self, database):
 
         return authenticated
 
-    def update_deployment(self, admindb):
-        props = admindb.command("isMaster")
-        if props.get("ismaster") == "isdbgrid":
-            self.deployment = MongosDeployment()
-        elif props.get("hosts"):
+    @staticmethod
+    def get_deployment(admindb):
+        options = admindb.command("getCmdLineOpts")['parsed']
```

Comment:
`getCmdLineOpts` is the runtime configuration of the mongo instance. Helpful to know whether the node is a mongos or mongod, if the mongod is in a shard, if it's in a replica set etc.

Reply:
Maybe add this as a code comment above this line?

```suggestion
        # getCmdLineOpts is the runtime configuration of the mongo instance. Helpful to know whether the node is
        # a mongos or mongod, if the mongod is in a shard, if it's in a replica set, etc.
        options = admindb.command("getCmdLineOpts")['parsed']
```